checker/src/org/checkerframework/checker/signedness/SignednessVisitor.java

```diff
@@ -91,6 +106,34 @@ private boolean maskIgnoresMSB(Kind maskKind, LiteralTree numBitsLit, LiteralTre
     }
 
     /**
+     * Given a casted right shift of the form {@code (type) (baseExpr >> numBits)} or {@code (type)
+     * (baseExpr >>> numBits)}, return true iff the expression's value is the same regardless of the
+     * type of right shift (signed or unsigned). This is true if the cast ignores the numBits most
+     * significant bits of the shift result -- that is, if the cast ignores all the new bits that
+     * the right shift introduced on the left.
+     *
+     * <p>For example, the function returns true for
+     *
+     * <pre>{@code (short) (myInt >> 16)}</pre>
+     *
+     * and for
+     *
+     * <pre>{@code (short) (myInt >>> 16)}</pre>
+     *
+     * because these two expressions are guaranteed to have the same result.
+     *
+     * @param numBitsLit the LiteralTree whose value is numBits
+     * @param bitDiff the difference between the widths of the type of the right shift (which is int
```

Comment:
I find this definition of the formal parameter confusing.  I think the interface would be easier to understand if this routine took the two bit widths and did the subtraction itself.  There doesn't seem to be much benefit to the client doing the subtraction instead of moving it into this routine.

Or, you could pass the types into this routine.

Or just inline the routine into the one place it is called, since it is only two lines long.

In any event, it seems odd to have such a confusing interface and so much repeated documentation to support a two-line method.